src/eureka/S3_data_reduction/plots_s3.py

```diff
@@ -1100,3 +1108,46 @@ def phot_2d_frame_diff(data, meta):
         plt.savefig(meta.outputdir + fname, dpi=250)
         if not meta.hide_plots:
             plt.pause(0.2)
+
+def stddev_profile(meta, n, m, stdevs, p7thresh):
+    """
+    Plots the difference between the data and optimal profile in units
+    of standard deviations.  The scale goes from 0 to p7thresh. (Fig 3506)
+
+    Parameters
+    ----------
+    meta : eureka.lib.readECF.MetaClass
+        The metadata object.
+    n : int
+        The current integration number.
+    m : int
+        The file number.
+    stdevs : 2D array
+        Difference between data and profile in standard deviations
+    p7thresh : int
+        X-sigma threshold for outlier rejection during optimal spectral
+        extraction
+
+    Notes
+    -----
+    History:
+
+    - 2022-12-29 Kevin Stevenson
+        Initial version
+    """
+    plt.figure(3506, figsize=(8, 4))
+    plt.clf()
+    cmap = plt.cm.viridis.copy()
+    plt.title(f'Std. Dev. from Optimal Profile - Integration {n}')
+    plt.imshow(stdevs, origin='lower', aspect='auto',
+               vmax=p7thresh, vmin=0, cmap=cmap,
+               interpolation='nearest')
+    plt.ylabel('Relative Pixel Position')
+    plt.xlabel('Relative Pixel Position')
+    plt.colorbar()
+    plt.tight_layout()
+    file_number = str(m).zfill(int(np.floor(np.log10(meta.num_data_files))+1))
+    int_number = str(n).zfill(int(np.floor(np.log10(meta.n_int))+1))
+    fname = (f'figs{os.sep}fig3506_file{file_number}_int{int_number}' +
+             f'_Std_Dev_Profile'+plots.figure_filetype)
+    plt.savefig(meta.outputdir + fname, dpi=200)
```

Comment:
Do you want to add a `plt.pause(0.2)` if `not meta.hideplots` like with the other plots?